docs/specification.md

```diff
@@ -1864,6 +3027,104 @@ data: {"statusUpdate": { /* TaskStatusUpdateEvent */ }}
+
+### 12.3. Header Transmission
+
+As specified in [Section 3.2.5 (Headers)](#325-headers), custom protocol bindings **MUST** document how headers are transmitted. The binding specification **MUST** address:
+
+1. **Transmission Mechanism**: The protocol-specific method for transmitting header key-value pairs
+2. **Value Constraints**: Any limitations on header values (e.g., character encoding, size limits)
+3. **Reserved Names**: Any header names reserved by the binding itself
+4. **Fallback Strategy**: What happens when the protocol lacks native header support (e.g., passing headers in metadata)
+
+**Example Documentation Requirements:**
+
+- **For native header support**: "Headers are transmitted using HTTP request headers. Header keys are case-insensitive and must conform to RFC 7230. Header values must be UTF-8 strings."
+- **For protocols without headers**: "Headers are serialized as a JSON object and transmitted in the request metadata field `a2a-headers`."
+
+### 12.4. Error Mapping
+
+Custom bindings **MUST**:
+
+1. **Map Standard Errors**: Provide mappings for all A2A-specific error types defined in [Section 3.2.2 (Error Handling)](#332-error-handling)
+2. **Preserve Error Information**: Ensure error details are accessible to clients
+3. **Use Appropriate Codes**: Map to protocol-native error codes where applicable
+4. **Document Error Format**: Specify the structure of error responses
+
+### 12.5. Streaming Support
+
+If the binding supports streaming operations:
+
+1. **Define Stream Mechanism**: Document how streaming is implemented (e.g., WebSockets, long-polling, chunked encoding)
+2. **Event Ordering**: Specify ordering guarantees for streaming events
+3. **Reconnection**: Define behavior for connection interruption and resumption
+4. **Stream Termination**: Specify how stream completion is signaled
+
+If streaming is not supported, the binding **MUST** clearly document this limitation in the Agent Card.
+
+### 12.6. Authentication and Authorization
+
+Custom bindings **MUST**:
+
+1. **Support Standard Schemes**: Implement authentication schemes declared in the Agent Card
+2. **Document Integration**: Specify how credentials are transmitted in the protocol
+3. **Handle Challenges**: Define how authentication challenges are communicated
+4. **Maintain Security**: Follow security best practices for the transport protocol
+
+### 12.7. Agent Card Declaration
+
+Custom bindings **MUST** be declared in the Agent Card:
+
+1. **Transport Identifier**: Use a clear, descriptive transport name
+2. **Endpoint URL**: Provide the full URL where the binding is available
+3. **Documentation Link**: Include a URL to the complete binding specification
+
+**Example:**
+```json
+{
+  "url": "wss://agent.example.com/a2a/websocket",
+  "preferredTransport": "WEBSOCKET",
+  "additionalInterfaces": [
+    {
+      "url": "wss://agent.example.com/a2a/websocket",
+      "transport": "WEBSOCKET"
+    }
+  ]
+}
```

Comment:
Looking at this made me realise that the top level "URL" field is redundant now. We should have just:

```
"interfaces": [
	{
		"url": "wss://agent.example.com/a2a/websocket",
		"supportsStreaming": True,
		"transport": "WEBSOCKET"
	},
	{
		"url": "https://...",
		"supportsStreaming": False,
		"transport": "JSONRPC"
	}
]
"preferredTransport": "WEBSOCKET"
```

"PreferredTransport" can then be looked up from the "interfaces" list. This way we only need to add additional info to the AgentInterface object like "streaming": true otherwise we need to add it at the top level and the transport level.

Reply:
I think we have to do this.  I have a question for you though.  How would feel if we said that interfaces was an ordered array and the first element in the list was preferred?  If we do that, then people can reference `/interfaces[0]/url` to get the preferred url, rather than have to go get the preferredTransport and then walk the array and match, and then we have to make sure there are not two elements in the array that have the same transport, etc.